### DISCUSSION TITLE
How to avoid the loaders of route re-running when a fetcher.Form submitted?

### DISCUSSION THREAD

**Discussion by BugMaker-Huang** [remix-run/remix#7329] [Q&A] [answered]
I have a API-Route, and a `fetcher.Form` to get data from its loader, when I submitted this form, all my loaders(include root) re-running, that makes me sad:(
I just want to get data from an API-Route with form and without refresh and navigation, how can I do that?

**Comment by sergiodxa** [comment, marked as answer]
You can use the [shouldRevalidate](https://remix.run/docs/en/1.19.3/route/should-revalidate)  function to control that behavior

**Reply by BugMaker-Huang** [reply]
Thanks, it looks helpful.
